src/Guard.php

```diff
@@ -417,14 +417,18 @@ protected function appendTokenToRequest(ServerRequestInterface $request, array $
     public function process(ServerRequestInterface $request, RequestHandlerInterface $handler): ResponseInterface
     {
         $body = $request->getParsedBody();
+        $headers = $request->getHeaders();
         $name = null;
         $value = null;
 
         if (is_array($body)) {
             $name = $body[$this->getTokenNameKey()] ?? null;
             $value = $body[$this->getTokenValueKey()] ?? null;
         }
-
+        if (($name == null || $value == null) && isset($headers[$this->getTokenNameKey()], $headers[$this->getTokenValueKey()])) {
```

Comment:
You do not need `isset` here. Next lines already checking token existence.

Reply:
agree, that was redundant, how about the new commit i did?

Reply:
You do NOT need to check if $header is an array since `ServerRequestInterface::getHeaders()` already return `string[][]`. If no header, it will be an empty `string`.
Here, you just need to check `$name == null || $value == null` 😄

Reply:
somehow this needs to be checked i think, cuz its a nested array, both headers and headers[tokennamekey][0]

Reply:
If `$headers` is not an array, it means you are using a bad PSR-7 provider. This package needs a compliant PSR-7 provider.

Following are equivalent:
```php
$name = $headers[$this->getTokenNameKey()][0] ?? null;
```
and
```php
$name = null;
if (isset($headers[$this->getTokenNameKey()]) && isset($headers[$this->getTokenNameKey()][0])) {
    $name = $headers[$this->getTokenNameKey()][0];
}
```